### PR TITLE
network: split configure hostname task out of network installation ta…

### PR DESCRIPTION
**Pull request by rvykydal**
…sk (#1757960)

Task for hostname configuration is useful for Inital setup.

### CODE REVIEW

File: pyanaconda/modules/network/installation.py [hunk 1]
```diff
@@ -33,10 +33,77 @@
...
+        :param hostname: static hostname
+        :type hostname: str
+        :param overwrite: overwrite existing configuration file
+        :type overwrite: bool
+        """
+        return _write_config_file(root, self.HOSTNAME_CONF_FILE_PATH,
+                                  "{}\n".format(hostname),
+                                  "Cannot write hostname configuration file",
+                                  overwrite)
+
```
Comment (jkonecny12) [inline pyanaconda/modules/network/installation.py]: I think you can inline this method to the run method.

File: pyanaconda/modules/network/network.py [hunk 2]
```diff
@@ -336,6 +336,19 @@ def install_network_with_task(self, overwrite):
         task.succeeded_signal.connect(lambda: self.log_task_result(task, root_path=conf.target.system_root))
         return task
 
+    def configure_hostname_with_task(self, overwrite):
+        """Configure hostname with an installation task.
+
+        :param overwrite: overwrite existing configuration
+        :return: a DBus path of an installation task
+        """
+
```
Comment (jkonecny12) [inline pyanaconda/modules/network/network.py]: Please remove this blank line.

File: tests/nosetests/pyanaconda_tests/module_network_test.py [hunk 3]
```diff
@@ -214,6 +213,20 @@ def install_network_with_task_test(self, devices_ignore_ipv6, publisher):
...
         self.network_module.log_task_result.assert_called_once()
 
+    @patch_dbus_publish_object
+    def configure_hostname_with_task_test(self, publisher):
+        """Test ConfigureHostnameWithTask."""
+        self.network_module._hostname = "my_hostname"
+
+        task_path = self.network_interface.ConfigureHostnameWithTask(
+            False,
+        )
```
Comment (jkonecny12) [inline tests/nosetests/pyanaconda_tests/module_network_test.py]: Please write this on one line.

File: pyanaconda/modules/network/installation.py (hunk 1 above)
Comment (rvykydal) [reply pyanaconda/modules/network/installation.py]: Done

File: pyanaconda/modules/network/network.py (hunk 2 above)
Comment (rvykydal) [reply pyanaconda/modules/network/network.py]: Fixed.

File: tests/nosetests/pyanaconda_tests/module_network_test.py (hunk 3 above)
Comment (rvykydal) [reply tests/nosetests/pyanaconda_tests/module_network_test.py]: Fixed.

**Pull request merged**